### ISSUE TITLE
Curly notation: if, else if, else for, else while, else do, else

### ISSUE THREAD

**Issue by StephanBijzitter** [eslint/eslint#3914]
It may be a bit against common convention, but that's what we have options for.
I would propose an option to treat `else for`, `else while` and `else do` as `else if` with regards to curly braces.

I have currently set up my rules as follows `"curly": [2, "all"]`:
```javascript
if (a) {

} else if (b) {

} else {
    for (var i = 0; i < c; i++) {

    }
}
```

But I would like to be able to write my code this way:
```javascript
if (a) {

} else if (b) {

} else for (var i = 0; i < c; i++) {

}
```
Realising that `else if` allows further statements, I do recognise the functional difference. However, as this is a difference that would result in syntax errors if done wrong, I believe it is not in the scope of eslint to handle this.

My proposal differs from the *braceless* `else` as follows:
```javascript
//bad
if (a) {

} else b();

//good
if (a) {

} else {
    b();
}

//good
if (a) {

} else while (b) {

}

//good
if (a) {

} else do {
    c();
} while (b);
```

To me, this increases readability as it allows for less indentation.

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by StephanBijzitter** [comment]
Just read the bot's help topic for proposing a new rule, it would be nice if it were to be included in the contributing message Github shows me when opening an issue.

I would be willing to develop this rule myself, given some pointers on how to proceed. The topic also introduced me to the plugins section, which would indeed be useful. I will try to create a plugin first to demo the rule before asking to include it in the core.

**Comment by nzakas** [comment]
We've found most people dont read the contribution message on new issues. It does currently link to the full contribution guidelines on our site.

If you want to try implementing this change, we can consider it. I'm skeptical it can be done in a way that 1) will make sense to most people, and 2) won't add a ton of complexity to the code. I could be wrong, though, so happy to entertain a pull request.

**Comment by gyandeeps** [comment]
@StephanBijzitter Are you still interested in putting out a PR for this rule? Since this has been sitting here for some months now.

**Comment by StephanBijzitter** [comment]
@gyandeeps Yeah I am, just have been away for some time! I'll pick up on my backlog starting next year :+1: 

https://github.com/Skelware/node-file-parser/blob/master/src/parsers/ini.js#L59

**Comment by gyandeeps** [comment]
@StephanBijzitter Sounds good. Thanks for the update.

**Comment by nzakas** [comment]
There hasn't been any progress here in four months, so closing.

ESLint team: if anyone wants to champion this change, please reopen.

**Comment by BYK** [comment]
This was an interesting suggestion. If @StephanBijzitter is still interested, I can assist him implement this.
